components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/subscription/registry/RegistrySubscriptionManager.java

```diff
@@ -144,63 +135,101 @@ public void addSubscription(Subscription subscription) throws EventBrokerExcepti
+        if (!topicName.startsWith(RegistryConstants.PATH_SEPARATOR)) {
+            resourcePath = resourcePath + RegistryConstants.PATH_SEPARATOR;
         }
 
         // this topic name can have # and * marks if the user wants to subscribes to the
         // child topics as well. but we consider the topic here as the topic name just before any
-        // special charactor.
+        // special character.
         // eg. if topic name is myTopic/*/* then topic name is myTopic
-        if (topicName.indexOf("*") > -1){
-            topicName = topicName.substring(0, topicName.indexOf("*"));
-        } else if (topicName.indexOf("#") > -1){
-            topicName = topicName.substring(0, topicName.indexOf("#"));
+        if (topicName.indexOf('*') > -1){
+            topicName = topicName.substring(0, topicName.indexOf('*'));
+        } else if (topicName.indexOf('#') > -1){
+            topicName = topicName.substring(0, topicName.indexOf('#'));
         }
 
-        resourcePath += topicName;
+        resourcePath = resourcePath + topicName;
 
-        if (!resourcePath.endsWith("/")) {
-            resourcePath += "/";
+        if (!resourcePath.endsWith(RegistryConstants.PATH_SEPARATOR)) {
+            resourcePath = resourcePath + RegistryConstants.PATH_SEPARATOR;
         }
 
-        resourcePath += EventBrokerConstants.EB_CONF_WS_SUBSCRIPTION_COLLECTION_NAME + "/" + subscriptionID;
+        resourcePath = resourcePath + EventBrokerConstants.EB_CONF_WS_SUBSCRIPTION_COLLECTION_NAME +
+                       RegistryConstants.PATH_SEPARATOR + subscriptionID;
         return resourcePath;
     }
 
     /**
-     * Get all the subscriptions. it get the subscription ids from the topic index and get
-     * relavent subscriptions.
-     * @return
-     * @throws org.wso2.carbon.event.core.exception.EventBrokerException
+     * Calculates the JMS subscription stored path for a WSSubscription using subscription id and the topic name
+     *
+     * @param subscriptionID the subscription ID
+     * @param topicName      the topic name
+     * @return the JMS subscription resource path for a subscription
      */
+    private String getJMSSubResourcePath(String subscriptionID, String topicName) {
+        String resourcePath = this.topicStoragePath;
+
+        // first convert the . to /
+        topicName = topicName.replaceAll("\\.", RegistryConstants.PATH_SEPARATOR);
+
+        if (!topicName.startsWith(RegistryConstants.PATH_SEPARATOR)) {
+            resourcePath = resourcePath + RegistryConstants.PATH_SEPARATOR;
+        }
+
+        // this topic name can have # and * marks if the user wants to subscribes to the
+        // child topics as well. but we consider the topic here as the topic name just before any
+        // special character.
+        // eg. if topic name is myTopic/*/* then topic name is myTopic
+        if (topicName.indexOf('*') > -1) {
+            topicName = topicName.substring(0, topicName.indexOf('*'));
```

Comment:
According to AMQP specification, we can have * anywhere.

AMQP specification 0-9-1. Page 27